doc/storage-backends.md

```diff
@@ -18,7 +18,12 @@ Nesting support                             | yes       | yes   | no    | no
 Restore from older snapshots (not latest)   | yes       | yes   | yes   | no
 Storage quotas                              | no        | yes   | no    | yes
 
-## Mixed storage
+With the implementation of the new storage api it is possible to use multiple
+storage drivers (e.g. BTRFS and ZFS) at the same time. Most of the sections
+below are only applicable to pre-storage-api LXD instances. They are kept here
+as references and marked with "pre-storage-api".
```

Comment:
We shouldn't keep deprecated documentation around, lets just update the different sections below to reflect how things work with the storage API.

Folks who are running pre-storage API LXD, should look at the doc for their release tag.